### PR TITLE
[SPARK-39661][SQL] Avoid creating unnecessary SLF4J Logger

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr follow up: https://issues.apache.org/jira/browse/SPARK-39538 https://github.com/apache/spark/pull/36930

### Why are the changes needed?
To improve performance.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Existing UT.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala
Comment (panbingkun) [inline core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala:60]: "with Logging" is more consistent and natural here.

Comment (mridulm) [reply core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala:60]: Given the `logger` exposed via `BlockStoreClient`, why use `Logging` here ?

**Review by mridulm — COMMENTED** [review]
Change from instance to static variable makes sense.
You can revert changes to `NettyBlockTransferService`.

**Comment by panbingkun** [comment]
> Change from instance to static variable makes sense. You can revert changes to `NettyBlockTransferService`.
Reason：Because nettyblocktransferservice is a singleton in every executor ？
Ok，i will revert nettyblocktransferservice.

**Comment by srowen** [comment]
Same in `BlockStoreClient` - can you change that too?

**Comment by panbingkun** [comment]
> Same in `BlockStoreClient` - can you change that too?

Updated.

**Comment by panbingkun** [comment]
> Hm, tests timed out. Can you retrigger? I'm sure it's not related, but good to see tests pass to be sure.

okay

File: common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java
Comment (LuciferYang) [inline common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java:45]: hmm... If `static final` is used, I think the variable name should be capitalized.

**Comment by LuciferYang** [comment]
@panbingkun test still timeout

File: common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java
Comment (srowen) [reply common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java:45]: Yeah go ahead and make it LOGGER and let's try tests one more time. I'm sure it's not related so would probably merge anyway.

Comment (LuciferYang) [reply common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java:45]: Please don't merge this, I tested the timeout case locally use follow command:

```
build/sbt "core/testOnly *BlockManagerBasicStrategyReplicationSuite"
```
the test is really hang

image

and I test this case in master, all passed.

Comment (srowen) [reply common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java:45]: Whoa, hm, I wonder how - some kind of locking issue now that the logger is shared?

File: sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/RowBasedKeyValueBatch.java
Comment (LuciferYang) [inline sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/RowBasedKeyValueBatch.java:51]: this change is ok

File: common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java
Comment (LuciferYang) [reply common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java:45]:
```
22/07/07 21:30:55.200 shuffle-client-39-1 WARN TransportChannelHandler: Exception in connection from localhost/127.0.0.1:52433
java.lang.IllegalAccessError: tried to access field org.apache.spark.network.shuffle.BlockStoreClient.logger from class org.apache.spark.network.netty.NettyBlockTransferService$$anon$3
        at org.apache.spark.network.netty.NettyBlockTransferService$$anon$3.onSuccess(NettyBlockTransferService.scala:185)
        at org.apache.spark.network.client.TransportResponseHandler.handle(TransportResponseHandler.java:196)
        at org.apache.spark.network.server.TransportChannelHandler.channelRead0(TransportChannelHandler.java:142)
        at org.apache.spark.network.server.TransportChannelHandler.channelRead0(TransportChannelHandler.java:53)
        at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
        at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
        at org.apache.spark.network.util.TransportFrameDecoder.channelRead(TransportFrameDecoder.java:102)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
        at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:722)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:658)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:584)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:496)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:995)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.lang.Thread.run(Thread.java:750)
```


@srowen @panbingkun  from above message, the `logger` in `BlockStoreClient` should be `public`

Comment (LuciferYang) [reply common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/BlockStoreClient.java:45]: I wonder why this is not a Scala compile time error

**Comment by srowen** [comment]
Ah OK, ignore my suggestion to change BlockStoreClient. I don't quite get why it compiled either. but I think an logger in the instance might be correct there.

**Comment by panbingkun** [comment]
> Ah OK, ignore my suggestion to change BlockStoreClient. I don't quite get why it compiled either. but I think an logger in the instance might be correct there.

BlockStoreClient's child class - NettyBlockTransferService, only be instantiated once in SparkEnv. code:
https://github.com/panbingkun/spark/blob/master/core/src/main/scala/org/apache/spark/SparkEnv.scala#L365-L367

so A JVM has only one NettyBlockTransferService object, change BlockStoreClient may not have any effect.

**Comment by panbingkun** [comment]
I will revert it!

**Review by LuciferYang — APPROVED** [review]
LGTM if GA Passed

**Pull request closed without merging**
